### DISCUSSION TITLE
Assistance Needed: Bypassing GitHub Verification for E2E Test Automation

### DISCUSSION THREAD

**Discussion by ksslbr** [community/community#145864] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello everyone,

I am working on an end-to-end test automation script for a platform that requires connecting to GitHub to import repositories. While my script successfully logs into GitHub, it triggers a "new device" detection and sends a verification code via email, which blocks the test flow from progressing.

I have already tried enabling 2FA, but this generates a code on the web page that must be entered into the mobile app, further complicating automation.

Is there any way to disable this verification or streamline the process specifically for conducting E2E tests?

Note: The automation relies on Playwright built into a testing platform where installing additional libraries is not allowed.

Thank you in advance for your help!

**Comment by hithubuserr007** [comment]
Use Personal Access Tokens (PATs) – Instead of logging in via the UI, authenticate using a PAT, which bypasses the need for device verification. You can generate a PAT in GitHub under Settings > Developer Settings > Personal Access Tokens.
- OAuth Authentication – If your platform supports OAuth, you can authenticate using an OAuth token instead of username/password authentication, which may help avoid triggering the verification step.
- GitHub Enterprise or Organization Settings – If you're working within an enterprise or organization, check if there are settings that allow trusted devices or IP whitelisting.
- Passkeys or SSH Authentication – GitHub supports passkeys and SSH authentication, which might be more automation-friendly than traditional login methods.
- Persistent Browser Sessions – If your automation framework allows it, try maintaining a persistent browser session to avoid triggering "new device" detection
